controller/api/destination/watcher/endpoints_watcher.go

```diff
@@ -1059,3 +1137,40 @@ func isValidSlice(es *discovery.EndpointSlice) bool {
 
 	return true
 }
+
+func setToServerProtocol(k8sAPI *k8s.API, address *Address, port Port) error {
+	servers, err := k8sAPI.Srv().Lister().Servers("").List(labels.Everything())
+	if err != nil {
+		return fmt.Errorf("failed to list Servers: %s", err)
+	}
+	for _, server := range servers {
+		selector, err := metav1.LabelSelectorAsSelector(server.Spec.PodSelector)
+		if err != nil {
+			return fmt.Errorf("failed to create Selector: %s", err)
+		}
+		if server.Spec.ProxyProtocol == opaqueProtocol && selector.Matches(labels.Set(address.Pod.Labels)) {
```

Comment:
can `address.Pod` be nil here for addresses not backed by a pod?  should we defend against a nil pointer deref?

Reply:
It shouldn't be as this is only called when endpoint's `TargetRef.Kind` is a pod. It probably wouldn't hurt to have a check at the top of the function to assert that, but it shouldn't be a situation we have to defend against often.

Reply:
`newPodRefAddress` will have already [errored](https://github.com/linkerd/linkerd2/blob/10650142bad3a64f4491c72241bd9a2dd5f67594/controller/api/destination/watcher/endpoints_watcher.go#L861) if the endpoint is not backed by a pod so it would be a redundant check here.

Reply:
still probably a good idea to be a bit defensive in case this gets refactored in the future or to document that as a precondition of this function.